src/app/src/main/kotlin/org/vaccineimpact/orderlyweb/controllers/UserController.kt

```diff
@@ -0,0 +1,22 @@
+package org.vaccineimpact.orderlyweb.controllers
+
+import org.vaccineimpact.orderlyweb.ActionContext
+import org.vaccineimpact.orderlyweb.Serializer
+import org.vaccineimpact.orderlyweb.models.AuthenticationResponse
+import org.vaccineimpact.orderlyweb.security.WebTokenHelper
+
+class UserController(context: ActionContext,
+                     private val tokenHelper: WebTokenHelper,
+                     private val serializer: Serializer)
+    : Controller(context)
+{
+    constructor(context: ActionContext) : this(context, WebTokenHelper.instance, Serializer.instance)
+
+    fun githubAuth(): AuthenticationResponse
```

Comment:
So if I've understood correctly, if the code gets this far, we've already confirmed that the user can authenticate via whatever auth method (secured at router level via config factory) so here we're happy to just issue to token and return in response. In with case, could it just be auth() rather than guthubAuth(). We should be able to work out the auth method from the header if we want to record it in the db etc.